### ISSUE TITLE
TOC generation

### ISSUE THREAD

**Issue by wzsRQQjaYN** [cfenollosa/bashblog#180]
Perl Markdown doesn't have inbuilt TOC generation, it's be great to implement one in bb.sh itself. If I find a compact solution I'll send a PR. till then any ideas on how to?

**Comment by wzsRQQjaYN** [comment]
There seem to be some ways to do this by using pandoc. but I think just for TOC introducing pandoc in the flow is against the whole minimalistic philosophy here. so we need to find a way within the bash script. and I think this can be done by some combinations of cat, grep, sed, and awk. I will soon send a PR if I do it.

**Comment by wzsRQQjaYN** [comment]
```
cat file.md | grep -E "^#{1,6} " | sed -E 's/(#+) (.+)/\1:\2:\2/g' | awk -F ':' '{gsub(/#/,"  ",$1);  gsub(/[0-9]./,"",$3); gsub(/[ ]/,"-",$3); print $1 "- [" $2 "](#" tolower($3) ")" }' | sed -r 'H;$!d;x;:a;/\n+\S/!s/(\n+) /\1/g;ta;s/.//'
```

This code generates a ToC out of a markdown file and prints to stdout. I've used some heuristics, perhaps someone could make it more general and then append it to the beginning (but after the title) of the same markdown file. then perhaps make this code into a function within the `bb.sh` script. 

I didn't want to spend time on this because I've found that the Pandoc version is way better than Gruber's Markdown.pl, e.g. footnotes, bibliography etc... although Pandoc is supposed to generate ToC automatically (using some appropriate switches, e.g. `--toc`) it doesn't seem to work, especially for markdown -> HTML conversion, the problem seems to be internal to pandoc in how the `--toc` works (it requires `--standalone` switch along with it, that returns the fully HTML page while the bashblog only requires the body part. so the solution instead is to create a modified.html template and use pass `--template=modified.html` to pandoc. (EDIT: for details checkout [wiki:Pandoc-as-bashblog's-markdown-compiler](https://github.com/cfenollosa/bashblog/wiki/Pandoc-as-bashblog's-markdown-compiler)

So, there's nothing to be done in `bb.sh` itself. all we need is a new Mardown compiler, and that's pandoc (with some settings).
